sources/org.osbuild.oci-archive

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/python3
+"""Fetch OCI image from a registry
+
+"""
+
+
+import os
+import sys
+import subprocess
+import uuid
+import tempfile
+
+from osbuild import sources
+
+
+SCHEMA = """
+"additionalProperties": false,
+"definitions": {
+  "item": {
+    "description": "The digest to fetch indexed their checksum",
+    "type": "object",
+    "additionalProperties": false,
+    "patternProperties": {
+      "sha256:[0-9a-f]{64}": {
+        "type": "object",
+        "additionalProperties": false,
+        "required": ["image"],
+        "properties": {
+          "image": {
+            "type": "object",
+            "additionalProperties": false,
+            "required": ["name"],
+            "properties": {
+              "name": {
+                "type": "string",
+                "description": "Name of the image (including registry)."
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+},
+"properties": {
+  "items": {"$ref": "#/definitions/item"},
+  "digests": {"$ref": "#/definitions/item"}
+},
+"oneOf": [{
+  "required": ["items"]
+}, {
+  "required": ["digests"]
+}]
+"""
+
+def download(items, cache):
+    for digest, item in items.items():
+        image = item["image"]
+        imagename = image["name"]
+
+        path = f"{cache}/{digest}"
+
+        if os.path.isfile(path):
+            continue
+
+        with tempfile.NamedTemporaryFile(prefix="tmp-copy-", dir=cache, mode="w") as tmpfile:
+            subprocess.run(["skopeo", "copy",
```

Comment:
Consider adding skopeo as a dependency to the README.

Reply:
It is not needed for the host, only for the build root, and that is documented in the stage docs.

Reply:
Actually christian said the sources are run on the host, so we should add this.